app/domain/repositories/message.go

```diff
@@ -3,6 +3,7 @@ package repositories
 import "github.com/limechain/hedera-eth-bridge-validator/app/persistence/message"
 
 type MessageRepository interface {
-	GetTransaction(txId, signature string) ([]message.TransactionMessage, error)
+	GetTransaction(txId, signature, hash string) (*message.TransactionMessage, error)
 	Create(message *message.TransactionMessage) error
+	GetByTransactionWith(txId string, txHash string) ([]message.TransactionMessage, error)
```

Comment:
Suggestion: 
`GetTransaction(txId, signature, hash string)` <- returns a single tx message
`GetTransactions(txId, txHash string)` <- returns an array of tx messages